app/models/person.rb

```diff
@@ -52,4 +66,18 @@ def links
   def details
     @content_item.content_item_data["details"]
   end
+
+  def ordered_previous_appointments
+    previous_roles = []
+    links["ordered_previous_appointments"].each do |previous_appointment|
+      if previous_appointment["links"]["role"].present?
+        previous_appointment["links"]["role"].each do |role|
```

Comment:
There [will actually only ever be one `role` here](https://github.com/alphagov/whitehall/blob/9291b03f5e986b894c35f5f8ffa6141216d0a506/app/presenters/publishing_api/role_appointment_presenter.rb#L28-L37) so I wouldn't worry about iterating over each one and just use `role = previous_appointment["links"]["role"].first`.

Reply:
Thanks for pointing that out. I'll update this :)